snuba/clickhouse/native.py

```diff
@@ -146,7 +147,7 @@ def __transform_result(self, result, with_totals: bool) -> Result:
 
     def execute(
         self,
-        query: str,
+        query: ClickhouseQuery,
         settings: Optional[Mapping[str, str]] = None,
         query_id: Optional[str] = None,
         with_totals: bool = False,
```

Comment:
We should move `with_totals` onto `ClickhouseQuery`, I think. Doesn't need to happen here, though.

I guess really, just about all of these parameters (except for `query` itself) are ClickHouse-specific and should probably be on the query class and not part of the method signature.

Reply:
added a todo